### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by dot-Eagle96**
Implemented some functionality and callbacks. No known bugs (as far as I know).

### CODE REVIEW

File: pycaw/callbacks.py [hunk 1]
```diff
@@ -64,6 +67,8 @@ class AudioSessionEvents(COMObject):
     -------
     Override the following method(s):
 
+    ### DOCSTRING MUST BE UPDATED WITH NEWER CONTENT ###
+
     def on_simple_volume_changed(self, new_volume, new_mute, event_context):
         Is fired, when the audio session volume/mute changed.
```
Comment (AndreMiras) [inline pycaw/callbacks.py:73]: Could you update accordingly?

File: pycaw/utils.py [hunk 2]
```diff
@@ -264,3 +272,33 @@ def GetAllDevices():
             if dev is not None:
                 devices.append(AudioUtilities.CreateDevice(dev))
         return devices
+
+    # Not sure that those should be here
+    @staticmethod
+    def GetDeviceEnumerator():
```
Comment (AndreMiras) [inline pycaw/utils.py:278]: Yes we can probably take it out

**Review by AndreMiras — COMMENTED** [review]
Pretty neat thanks!
I realized the CI wasn't running on pull requests lately, I've just fixed that.
Would you mind rebasing?
```
git fetch --all
git rebase upstream/develop # given this remote is called upstream
```
Last thing is would you mind being a bit more descriptive in your commit message.
Maybe something like:
> Improved audioclient, audiopolicy and mmdeviceapi support

You can edit a commit message with:
```
git commit --amend
# then `push --force`
```

**Comment by AndreMiras** [comment]
Thanks a mil, that looks pretty good!
Do you mind fixing the linter issues returned by the CI?
Running black should fix most of it.
```
tox -e lint-format
```
Then you can verify if they're some missing linting issues with:
```sh
tox -e lint-check
```
Also test coverage decreased, ideally we would write test for the new features, but I understand that could be challenging sometimes so feel free to address or not.
Yes you're right about a clear state/roadmap of missing features even though I'm not particularly looking to support them all, but more based on usage/demand

**Comment by dot-Eagle96** [comment]
Linter issues should be fixed now.
As for the new tests, I've never done such a thing before, so I need time to figure out how to implement them.

**Comment by AndreMiras** [comment]
Merged, thanks for addressing :pray: 
Feel free to look into testing if you feel like it.
Ideally we would want to get back to at least 70% of coverage

**Pull request merged**